### ISSUE TITLE
Invalid assignment target should not be allowed in a for-of iteration

### ISSUE THREAD

**Issue by ariya** [eslint/espree#170]
Test case:

```js
var espree = require('espree');
var options = { ecmaFeatures: { forOf: true } };
var ast = espree.parse('for (this of foo){}', options);
```

Actual: a syntax tree.

Expected: an error.

**Comment by xjamundx** [comment]
For reference acorn says this:

```
> require('./dist/acorn').parse('for (this of foo){}', {ecmaVersion: '6'})
SyntaxError: Assigning to rvalue (1:5)
    at Parser.pp.raise (/Users/jamuferguson/dev/acorn/dist/acorn.js:975:13)
    at Parser.pp.toAssignable (/Users/jamuferguson/dev/acorn/dist/acorn.js:1101:14)
    at Parser.pp.parseForStatement (/Users/jamuferguson/dev/acorn/dist/acorn.js:1805:10)
    at Parser.pp.parseStatement (/Users/jamuferguson/dev/acorn/dist/acorn.js:1693:19)
    at Parser.pp.parseTopLevel (/Users/jamuferguson/dev/acorn/dist/acorn.js:1653:21)
    at Parser.parse (/Users/jamuferguson/dev/acorn/dist/acorn.js:1623:17)
    at Object.parse (/Users/jamuferguson/dev/acorn/dist/acorn.js:937:44)
    at repl:1:25
    at REPLServer.defaultEval (repl.js:132:27)
    at bound (domain.js:254:14)
```

**Comment by nzakas** [comment]
Fixed in 3.0.0.
